### PR TITLE
Initial push for CSPRNG Support

### PR DESCRIPTION
**Pull request by saru95**
@panzone This is the initial push for the `Random` module. Some things I found out and had doubts about:
 
1. `RAND_bytes` takes care of seeding itself based on the best system-based seed available. So, no external seeding is required.
2. Calling the `RAND_poll` or `RAND_seed` is a bad practice since `RAND_bytes` already chooses the best possible option.
3. Do we need to support user-specified seed? If yes, only integers or CryptoBuffer too?
4. Do we also need a random number generator for practical use such as `r.randomRange(0, 10)` which generates random numbers between 0 to 10?

To test:

```chpl
use Crypto;
use Crypto;

var r = new Random();
var rbuff = r.randomize(25); // 25 is the number of random elements we need
writeln(rbuff.toHex());
```

### CODE REVIEW

**Comment by panzone** [comment]
While I think that we should try to provide a way to manually seed the generator, it' s quite difficult to do it in a OO environment because the OpenSSL generator is "global" (so seeding in an object constructor will reseed the state of all the other CryptoRandom objects). Unless you have some ideas...

(No, we don't need to provide a range function).

File: Crypto.chpl [hunk 1]
```diff
@@ -96,4 +99,11 @@ module Crypto {
     }
   }
 
+  class Random {
```
Comment (panzone) [inline Crypto.chpl]: Chapel already has a Random module, we can't use this name.

File: Crypto.chpl [hunk 2]
```diff
@@ -96,4 +99,11 @@ module Crypto {
...
+    proc randomize(buffLen: int) {
```
Comment (panzone) [inline Crypto.chpl]: randomize doesn't give the idea of a new CryptoBuffer in my opinion. Maybe makeRandomBuffer ?

File: CryptoSupport/randomSupport.chpl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require "openssl/rand.h";
+require "CryptoSupport/CryptoUtils.chpl";
+
+module randomSupport {
```
Comment (panzone) [inline CryptoSupport/randomSupport.chpl]: Better CryptoRandomSupport (because RandomSupport is already a Chapel class).

**Comment by saru95** [comment]
@panzone There is no way to provide a custom seed value for simplicity. Even during manual seeding, OpenSSL takes the value from the respective entropy it considers.

File: CryptoSupport/randomSupport.chpl [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+  extern proc RAND_bytes(buf: c_ptr(c_uchar), num: c_int): c_int;
+  extern proc RAND_seed(const buf: c_void_ptr, num: c_int);
+
+  proc randomize(buffLen: int) {
+    var buff: [0..(buffLen - 1)] uint(8);
+    RAND_bytes(c_ptrTo(buff): c_ptr(c_uchar), buffLen: c_int);
```
Comment (panzone) [inline CryptoSupport/randomSupport.chpl]: Oh yes, almost forgot. You need to check the returned value of RAND_bytes to be sure that everything went correctly.

Comment (saru95) [inline CryptoSupport/randomSupport.chpl]: Yep. I'll do that. Other changes you want me to make or add before merging?

**Comment by panzone** [comment]
Remember that you have to update the examples and, if possible, adding some new ones.

**Comment by saru95** [comment]
```chpl
use Crypto;
use Crypto;
 
 var r = new CryptoRandom();
 var rBuff = r.createRandomBuffer(20);
 writeln(rBuff.toHex());
```

**Pull request merged**